dashboard/app/controllers/api/v1/pd/teacher_applications_controller.rb

```diff
@@ -2,8 +2,27 @@ class Api::V1::Pd::TeacherApplicationsController < ApplicationController
   authorize_resource class: 'Pd::TeacherApplication'
 
   def index
-    @teacher_applications = ::Pd::TeacherApplication.all
-    render json: @teacher_applications.map{|a| JSON.parse(a.application).merge({user_id: a.user_id})}
+    @teacher_applications = ::Pd::TeacherApplication
+    if params[:after_id]
+      @teacher_applications = @teacher_applications.where('id > ?', params[:after_id])
+    end
+
+    expanded_results = @teacher_applications.all.map do |teacher_application|
+      application_json = JSON.parse(teacher_application.application)
+      school = School.find(application_json['school'])
+      school_district = SchoolDistrict.find(application_json['school-district'])
+
+      application_json.merge({
+        id: teacher_application.id,
+        userId: teacher_application.user_id,
+        timestamp: teacher_application.created_at,
+        schoolName: school.name,
+        schoolDistrictName: school_district.name,
+        regionalPartner: school_district.regional_partner.try(:name)
+      }).stringify_keys
+    end
+
+    render json: expanded_results
```

Comment:
Just to be clear, this returns an array of json and thats what we want here, right?